git/operations.go

```diff
@@ -276,35 +276,94 @@ func changed(ctx context.Context, workingDir, ref string, subPaths []string) ([]
 	return splitList(out.String()), nil
 }
 
-func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
-	if trace {
-		print("TRACE: git")
-		for _, arg := range args {
-			print(` "`, arg, `"`)
+// Whilst debugging or developing, you may wish to filter certain git commands out of the logs.
+func exemptedTraceGitCommand(command string) bool {
```

Comment:
I still don't think this needs to be on its own. It's not complicated enough to warrant having its own function, or used in more than one place, or tested (and needn't be).

You could move the `exemptedCommands` up to the package scope and put them beside `trace` (and rename them, since they will be out of context), since people might want to alter them at the same time as `trace`.

Reply:
just to be clear @squaremo you told me to extract it into a func, that's why I did.

![Screenshot_20190326_094534](https://user-images.githubusercontent.com/15232461/55001865-e87cb680-4fab-11e9-9d50-091df86cb756.png)

If we're on the topic I would prefer to just use a simple `contains` function, but that has already been added to [`aws.go`](https://github.com/weaveworks/flux/blob/1521741297657c38b9319dd1deb8f88cb1b72bcf/registry/aws.go#L38) and [`compat.go`](https://github.com/weaveworks/flux/blob/1521741297657c38b9319dd1deb8f88cb1b72bcf/remote/rpc/compat.go#L66) and this would make the third place this is copied in the project, that's why it's its own thing in the first place.  I think the semantics of
```go
traceExcemption := contains([]string{
        "config",
    }, args[0])
```
are very clear, but anyway... I'll happily change to match your latest request, no problemo.

Reply:
> you told me to extract it into a func

I meant the whole machinery, not just the trivial string-in-slice predicate.

Reply:
ok, well no worries.  I just mentioned it because I want you to understand that the only reason it was like that was because I was following your instructions as closely as possible.

thanks :)